src/Stratis.Bitcoin/Base/TimeSyncBehavior.cs

```diff
@@ -255,30 +258,41 @@ public bool AddTimeData(IPAddress peerAddress, TimeSpan offsetSample, bool isInb
         /// <remarks>
         /// The caller of this method is responsible for holding <see cref="lockObject"/>.
         /// <para>
-        /// The function takes a single copy of each inbound sample and combines them with <see cref="OutboundToInboundWeightRatio"/>
+        /// The function takes a single copy of each inbound sample and combines them with <see cref="OffsetWeightSecurityConstant"/>
         /// copies of outbound samples.
         /// </para>
         /// <para>
-        /// We require to have at least <see cref="MinUnweigtedSamplesCount"/> unweighted samples to change the value of <see cref="timeOffset"/>.
+        /// When there are many more inbound samples than unweighted outbound, which could be the case
+        /// in a malicious attack, the security is still maintained by using a dynamic multiplier on the outbound samples that maintains that weighting.
+        /// </para>
+        /// <para>
+        /// We require to have at least <see cref="MinOutboundSampleCount"/> outbound samples to change the value of <see cref="timeOffset"/>.
         /// </para>
         /// </remarks>
         private void RecalculateTimeOffsetLocked()
         {
             this.logger.LogTrace("()");
 
-            // All samples are formed of one copy of each inbound sample.
-            // And N copies of each outbound sample.
-            int unweightedSamplesCount = this.inboundTimestampOffsets.Count + OutboundToInboundWeightRatio * this.outboundTimestampOffsets.Count;
-
-            if (unweightedSamplesCount >= MinUnweightedSamplesCount)
+            if (this.outboundTimestampOffsets.Count != 0 && this.outboundTimestampOffsets.Count >= MinOutboundSampleCount)
```

Comment:
just `this.outboundTimestampOffsets.Count >= MinOutboundSampleCount` is enough

Reply:
agree